pkg/models/error.go

```diff
@@ -158,3 +184,30 @@ func (e *BaseError) Details() map[string]string {
 func (e *BaseError) Code() ErrorCode {
 	return e.code
 }
+
+// HTTPStatusCode is a method that returns the httpStatusCode field of BaseError.
+// If no specific HTTP status code has been set, it returns 0.
+// This method can be used to retrieve the HTTP status code associated with the error,
+// which is useful when translating the error into an HTTP response.
+func (e *BaseError) HTTPStatusCode() int {
+	return e.httpStatusCode
+}
```

Comment:
this method should return `e.httpStatusCode` if it is not 0, otherwise it should call `inferHttpStatusCode`

Reply:
Acknowledged fixing it